### PR TITLE
 QA: segwit.py: s/find_unspent/find_spendable_utxo/

### PR DESCRIPTION
**Pull request by jtimon**
Separated from #8994 
It was found out testing that PR but I think this fix should be done even without #8994 the fix is not necessary by luck. Unless I'm missing something.

### CODE REVIEW

**Comment by sipa** [comment]
I don't think it's obvious that a function with that name would only return spendable UTXOs. If there is a use case for it, perhaps make it a boolean argument require_spendable?

**Review by promag — COMMENTED** [review]
I would say to extend the scope to add `minimumSumAmount=min_value` and `maximumCount=1` to `listunspent` options BUT the result could be an unspendable unspent. I guess `listunspent` could have a new option to filter server side.

Also, callers assume this call always work, it should throw after the loop?

Agree with @sipa.

**Comment by jtimon** [comment]
Is there any use case where the function is used to return unspendable outputs? By defintion, if they're unspendable they're not part of the utxo, right?
I don't quite get why, but changing the genesis block of regtest (ie using the custom regtest proposed in #8994 ) makes this function return unspendable utxos and thus make segwit.py fail.
Or in other words, I don't quite get why the function never randomly returns an unspendable utxo and fails without #8994. 

Perhaps this doesn't make sense without #8994, but I don't see the point in adding new parameters to the function that aren't used for any test.

> Also, callers assume this call always work, it should throw after the loop?

Yeah, that makes sense to me.

**Comment by sipa** [comment]
@jtimon Ah, I see the confusion. Spendable here means you have the private keys and scripts necessary to spend them. Watch-only addresses can result in non-spendable UTXOs being listed.

**Comment by jtimon** [comment]
I see. So would it be fine to add the require_spendable argument and default to True?
I still don't see the point until some test actually needs to call it with False, but I don't mind adding it.

File: test/functional/segwit.py [hunk 1]
```diff
@@ -70,7 +70,7 @@ def getutxo(txid):
 
 def find_unspent(node, min_value):
     for utxo in node.listunspent():
```
Comment (MarcoFalke) [inline test/functional/segwit.py]: nit: Can use `query_options={"minimumAmount": min_value}`

File: test/functional/segwit.py [hunk 2]
```diff
@@ -70,7 +70,7 @@ def getutxo(txid):
...
-        if utxo['amount'] >= min_value:
+        if utxo['amount'] >= min_value and utxo['spendable']:
             return utxo
```
Comment (MarcoFalke) [inline test/functional/segwit.py]: Should `throw` for clarity.

File: test/functional/segwit.py [hunk 3]
```diff
@@ -70,7 +70,7 @@ def getutxo(txid):
 
 def find_unspent(node, min_value):
```
Comment (MarcoFalke) [inline test/functional/segwit.py]: nit: Rename to `find_spendable_output` or something.

**Review by maflcko — COMMENTED** [review]
utACK d01b06991a4e10b2cb95b25bfcddcddc58ed0738. Some style nits.

**Comment by MarcoFalke** [comment]
@jtimon Are you still working on this? Otherwise I will close

File: test/functional/segwit.py (hunk 3 above)
Comment (jtimon) [reply test/functional/segwit.py]: That would make the PR much more disruptive, is it worth it?
Also, if I add the require_spendable param, I don't think it makes sense to rename it to that.

File: test/functional/segwit.py (hunk 1 above)
Comment (jtimon) [reply test/functional/segwit.py]: Not sure I understand the suggestion.

File: test/functional/segwit.py (hunk 2 above)
Comment (jtimon) [reply test/functional/segwit.py]: Yeah, @promag pointed this out already and I said I would do it. Then I ask a question that nobody answered.

Comment (promag) [reply test/functional/segwit.py]: I'd say either add the new arg or rename the function and add the missing throw.

**Comment by jtimon** [comment]
@MarcoFalke I'm on vacation and I didn't saw your nits.
Does this have chances to be merge if I add the require_spendable argument and default to True?
IMO I would just do the throw thing and leave it as it is without require_spendable, but as said I don't strongly care, my main goal is making #8994 one commit shorter.

File: test/functional/segwit.py (hunk 3 above)
Comment (MarcoFalke) [reply test/functional/segwit.py]: > much more disruptive.

It is used in less than 5 places.

Changing the function name when the functionality changes aims review and prevents silent merge conflicts/bugs.

File: test/functional/segwit.py (hunk 1 above)
Comment (MarcoFalke) [reply test/functional/segwit.py]: `listunspent` takes named arguments according to the documentation, so you could add the named argument and remove some code from the python script instead.

**Comment by MarcoFalke** [comment]
@jtimon No rush. Feel free to revisit next year. ;)

**Comment by jtimon** [comment]
Hopefully fixed all nits.
Either I squash before merge or a ditch the last commit, opinions welcomed. I think I prefer the former but no strong opinion.
Rebased because why not?

File: test/functional/segwit.py [hunk 4]
```diff
@@ -68,11 +68,13 @@ def getutxo(txid):
...
 
-def find_unspent(node, min_value):
-    for utxo in node.listunspent():
-        if utxo['amount'] >= min_value:
+def find_spendable_output(node, min_value):
+    for utxo in node.listunspent(query_options={'minimumAmount': min_value}):
+        if utxo['spendable']:
             return utxo
 
+    raise AssertionError("Unspent output higher than %s not found." % min_value)
```
Comment (promag) [inline test/functional/segwit.py]: Nit, equal or higher.
Nit, remove `.`.

File: test/functional/segwit.py [hunk 5]
```diff
@@ -68,11 +68,13 @@ def getutxo(txid):
     utxo["txid"] = txid
     return utxo
 
-def find_unspent(node, min_value):
-    for utxo in node.listunspent():
-        if utxo['amount'] >= min_value:
+def find_spendable_output(node, min_value):
```
Comment (promag) [inline test/functional/segwit.py]: Nit, s/output/utxo.

**Review by promag — COMMENTED** [review]
Nits, don't mind ignoring them. +1 squash.

**Comment by jtimon** [comment]
Force pushed to fix newest nits.

**Comment by MarcoFalke** [comment]
utACK 305c345cc16c21160f07459156b36bd71e86395e

**Comment by MarcoFalke** [comment]
Needs rebase

**Comment by jtimon** [comment]
Rebased

**Comment by MarcoFalke** [comment]
re-utACK 9bb59cf

**Comment by jnewbery** [comment]
Tested ACK 9bb59cf7baea0b33d97ef0c0eeee88a8b104be3a

**Pull request merged**